### PR TITLE
Update to 3.9 and use secrets manager

### PR DESCRIPTION
**Pull request by codygordon**

### CODE REVIEW

File: settings.py.template [hunk 1]
```diff
@@ -1,11 +1,4 @@
 import os
```
Comment (sjwmoveon) [inline settings.py.template]: We should probably remove this file entirely, if we're switching to secrets manager.

File: validate_key.py [hunk 2]
```diff
@@ -21,9 +20,13 @@ def main(args):
         return {'valid': False}
     [key_created, age, source, campaign, hash] = args.KEY.split('.')
     m = hashlib.sha256()
+
+    script_settings = get_secret('ak-partner-rsvp')
+    secret = script_settings['SECRET']
```
Comment (sjwmoveon) [inline validate_key.py]: Can we name this something a little more descriptive?

**Review by sjwmoveon — CHANGES_REQUESTED** [review]

File: validate_key.py (hunk 2 above)
Comment (codygordon) [reply validate_key.py]: changed to `key_hash_secret`

File: settings.py.template (hunk 1 above)
Comment (codygordon) [reply settings.py.template]: ah yeah, done!

File: export_rsvps.py [hunk 3]
```diff
@@ -3,69 +3,58 @@
...
+                u.zip, MIN(a.created_at) AS action_datetime, MAX(s.role) AS role
+            FROM {db_schema}.core_user u
+            JOIN {db_schema}.events_eventsignup s ON s.user_id = u.id
+            JOIN {db_schema}.events_event e ON e.id = s.event_id
+            JOIN {db_schema}.events_campaign c ON c.id = e.campaign_id
+            LEFT JOIN {db_schema}.core_action a ON (
+                a.page_id = s.page_id
+                AND a.user_id = u.id
+            )
+            WHERE c.name = '{key.get('campaign', '')}'
```
Comment (sjwmoveon) [inline export_rsvps.py]: This was intentionally not included before - by passing it in as a parameter it's escaped, preventing issues with SQL injection.

File: validate_key.py [hunk 4]
```diff
@@ -1,37 +1,40 @@
...
 
-REQUIRED_ARGS = ['KEY', 'MAX_AGE', 'SECRET']
+REQUIRED_ARGS = ['KEY']
 
 
-def main(args):
+def main(args, script_settings={}):
+    if not script_settings:
+        script_settings = get_secret('ak-partner-rsvp')
+    key_hash_secret = script_settings['KEY_HASH_SECRET']
```
Comment (sjwmoveon) [inline validate_key.py:18]: Not sure this is really needed since it's only used once

**Review by sjwmoveon — CHANGES_REQUESTED** [review]

File: export_rsvps.py (hunk 3 above)
Comment (codygordon) [reply export_rsvps.py]: woops, will revert

File: validate_key.py (hunk 4 above)
Comment (codygordon) [reply validate_key.py:18]: sorry, do you mean the hash secret or something else?

Comment (sjwmoveon) [reply validate_key.py:18]: Yes, the hash secret.

**Review by sjwmoveon — APPROVED** [review]

**Pull request merged**